spec/controllers/api/v1/accounts/macros_controller_spec.rb

```diff
@@ -0,0 +1,133 @@
+require 'rails_helper'
+
+RSpec.describe 'Api::V1::Accounts::MacrosController', type: :request do
+  let(:account) { create(:account) }
```

Comment:
lets add the spec for an agent here and make sure only the visible macros are returned.
or that the call is made to self.with_visibility in macros model